From 8e7b02d972e605ed4906e0575c57f0a56ce58f62 Mon Sep 17 00:00:00 2001
From: Auke Kok <auke-jan.h.kok@intel.com>
Date: Fri, 2 Jun 2017 13:38:58 -0700
Subject: [PATCH 6/7] Delay writing out the chart result.

If the defaults in the config file are too short,
/run may not be mounted yet in time, it's better to just always
wait 1 second before writing out things.
---
 src/bootchart.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/bootchart.c b/src/bootchart.c
index 69c7e13..bee0335 100644
--- a/src/bootchart.c
+++ b/src/bootchart.c
@@ -480,6 +480,7 @@ int main(int argc, char *argv[]) {
                 t = time(NULL);
                 r = strftime(datestr, sizeof(datestr), "%Y%m%d-%H%M", localtime(&t));
                 assert_se(r > 0);
+                sleep(1);
 
                 snprintf(output_file, PATH_MAX, "%s/bootchart-%s.svg", arg_output_path, datestr);
                 of = fopen(output_file, "we");
-- 
2.13.0

